src/client/api/pythonApi.ts

```diff
@@ -352,16 +357,7 @@ export class InterpreterService implements IInterpreterService {
         if (isCI) {
             promise
                 .then((items) => {
-                    const current = interpreterCacheForCI.get(resource?.toString() || '');
-                    const itemToStore = items;
-                    if (
-                        current &&
-                        (itemToStore === current || JSON.stringify(itemToStore) === JSON.stringify(current))
-                    ) {
-                        return;
-                    }
-                    interpreterCacheForCI.set(resource?.toString() || '', itemToStore);
-                    traceInfo(`Interpreter list for ${resource?.toString()} is ${JSON.stringify(items)}`);
+                    traceInfo(`Interpreter list for ${resource?.toString()} is ${items.map((i) => i.path).join('\n')}`);
```

Comment:
Wait, isn't this a bug now
If the first call returns empty and second call returns non-empty that's a bug
I guess that's what you're trying to figure out right now.

Reply:
Yeah wanted to make sure this wasn't masking stuff. It was weird that it was caching only during CI.

Reply:
Ther'es no caching, if you check this out, its only for logging purposes.
I.e. the cache is only used for logging, not used in run time

Reply:
Ah right, it's just using the cache to skip logging.

Reply:
I actually want it to log every time to make sure we log each call. That might indicate the second call fails when the first passes.

Reply:
You might want to gate this logging with `traceInfoIf`
It should be smart to log only the first time (as we need this logging on user machines)
But do'nt want all subsequent calls to `getInterpreters` else we end up logging way too much (i think this gets logged in the output channel, and adds a lot of noise, as its a massive blob of JSON)

Reply:
It's already under an if (isCI) statement for the logging.

And I changed it to just log the path. That seems sufficient.